Source/Controllers/Preferences/Panes/SPFilePreferencePane.m

```diff
@@ -58,18 +71,46 @@ - (instancetype)init
 	return self;
 }
 
+- (IBAction)doubleClick:(id)sender {
+
+    SPLog(@"clickedColumn = %li", (long)fileView.clickedColumn);
+    SPLog(@"clickedRow = %li", (long)fileView.clickedRow);
+
+    if(fileView.clickedColumn >= 0 && fileView.clickedRow >= 0){
+        NSString *fileName = [fileNames safeObjectAtIndex:fileView.clickedRow];
+        SPLog(@"fileName: %@", fileName);
+        if(fileName != nil){
+
+            PanelOptions options = {
+                .title = @"Choose stale file",
+                .canChooseFiles = YES,
+                .canChooseDirectories = YES,
+                .allowsMultipleSelection = NO,
```

Comment:
Probably could allow multiple selection - we should just iterate through the files selected so if some user selected all of their files at once it should be okay, I believe

Reply:
The double click event is sent three times, once for no reason, and once for each file selected. At the moment it pops up one window with the first file's directory, then instantly changes to the next file. I'll try the pre-selection stuff above.

Reply:
You can select multiple files from the same directory as usual though.

The fileView does have a property called `selectedRowIndexes` that we could iterate through maybe.

Reply:
Ok that works, but pops up a panel for each file. Would be nice to chain them...

Reply:
OK, got them chained. Will make a little vid. Just want to prevent selection of non-stale items first.

Reply:
@Jason-Morcos @Kaspik What do you think of this?


https://user-images.githubusercontent.com/1576190/103536688-f3c57b80-4ecd-11eb-83bb-7df8ae25215b.mp4

Reply:
Super awesome! Could clicking "yes" in the initial "you got stale bookmarks" dialog pop open prefs and _immediately_ start the chain through with the file navigator? That way the user doesn't even need to know to double click the red items.
Also, might want to set the text prompt in the file navigator when restoring stale bookmarks (something like "please re-select the file "id.rsa" in order to restore Sequel Ace's access."

Reply:
> Also, might want to set the text prompt in the file navigator when restoring stale bookmarks (something like "please re-select the file "id.rsa" in order to restore Sequel Ace's access."

You mean on this screen?

![Screenshot 2021-01-05 at 5 35 36 PM](https://user-images.githubusercontent.com/1576190/103630484-aeac5280-4f7c-11eb-9032-2abcbfabd075.png)

Reply:
> Could clicking "yes" in the initial "you got stale bookmarks" dialog pop open prefs and _immediately_ start the chain through with the file navigator? That way the user doesn't even need to know to double click the red items.

I'm not sure I like that workflow, I'd like to see the files. I can try it, see what it feels like.

Reply:
It's a bit tricky, but looks like this... what do you think @Jason-Morcos @Kaspik 

https://user-images.githubusercontent.com/1576190/103640329-c4c10f80-4f8a-11eb-8e86-0dc650f90cb0.mp4

Reply:
Definitely tricky. I'm manually setting the selectedRows index set with the indexes of the stale files. Then trying to maintain them. Having them double click is going to be less buggy...

Reply:
I like the file browser prompt but I agree with you on the immediate file browser not being the way to go. It is nice to see the list of files first, and definitely don't want to do anything that's liable to be buggy just to skip a double click!

Reply:
A bit more work and:


https://user-images.githubusercontent.com/1576190/103661989-5ab86280-4faa-11eb-851c-3c168ee498cd.mp4

Reply:
TODO:

// what if the user clicks cancel, then double clicks just one file?
    // or different files?

Reply:
Hmmm not sure what to do for the to-do. My hunch for hitting cancel is to not chain the process then. And for different files chain whatever ones are selected together?

Reply:
> I like the file browser prompt but I agree with you on the immediate file browser not being the way to go. It is nice to see the list of files first, and definitely don't want to do anything that's liable to be buggy just to skip a double click!

so should I revert the code to not start the chaining of displaying the files?

Reply:
@Kaspik do you have any thoughts?

With the latest improvements you made I could go either way

Reply:
The most simple and understandable way :) I'm not following this really, sorry

Reply:
> @Kaspik do you have any thoughts?
> 
> With the latest improvements you made I could go either way

Let me see if I can get the cancel workflow to work, then I think it's good. They can click cancel and see what's what if they want

Reply:
OK, here's the flow for a user who doesn't click cancel, followed by a user that does, then selects files manually:


https://user-images.githubusercontent.com/1576190/103802187-b9084280-5089-11eb-97b9-4b4a6f06624e.mp4

Reply:
And, clicking cancel, and selecting multiple files, non-contiguous:

https://user-images.githubusercontent.com/1576190/103802365-f8cf2a00-5089-11eb-8ecc-c91c9eecc8d4.mp4

Reply:
And if a user clicks cancel at the initial popup, then opens file prefs:

https://user-images.githubusercontent.com/1576190/103803065-03d68a00-508b-11eb-8003-84961063191b.mp4

Reply:
Looks really really great!! I think it looks good to go